pkg/system/phase2_creating.go

```diff
@@ -607,7 +607,11 @@ func (r *Reconciler) ReconcileIBMCredentials() error {
 	// Currently IBM Cloud is not supported by cloud credential operator
 	// In IBM Cloud, the COS Creds will be provided through Secret.
 	r.Logger.Info("Running in IBM Cloud")
-	r.IsIBMCloud = true
+	util.KubeCheck(r.IBMCloudCOSCreds)
```

Comment:
You assume that the secret will be existent during phase2.
We just need to be sure that the IBM deployment scripts that deploy OCP on top of IBM will actually create it in time.
Did you verify that?

Reply:
Yes . Verified the secret creation part.